### ISSUE TITLE
Unable to use Steam Play for games with native ports

### ISSUE THREAD

**Issue by Lyle-Tafoya** [ValveSoftware/steam-for-linux#5638]
#### Your system information

* Steam client version (build number or date): August 22, 2018
* Distribution (e.g. Ubuntu): Arch Linux
* Opted into Steam client beta?: Yes
* Have you checked for system updates?: Yes

#### Please describe your issue in as much detail as possible:
Some games have native Linux builds, but have various problems or are missing some features (ie. VR support, controller support, online play, etc...). For these and other reasons, it is sometimes desirable to play the Windows version of a game, even if a native build is provided. Currently, games with a native build are not able to be installed via Steam Play and can only installed and run natively. I propose that Steam Play functionality should also be unlocked for games with native builds. Ideally, users would be able to install both the native and steam play versions side by side. However, I'd settle for picking one or the other.

#### Steps for reproducing this issue:

1. Find a game that supports both Linux and Windows and attempt to launch it via Steam Play

**Comment by Despruk** [comment]
Many good use cases mentioned in the Proton thread
#https://github.com/ValveSoftware/Proton/issues/52

**Comment by Gibbz** [comment]
Yes id love to see something like this also.
For example COH2 doesnt support cross platform play, so Id like to have access to both native version and windows version.

**Comment by Ryujinra** [comment]
Yes, there are also titles such as The Banner Saga, which advertise Linux support on the store page but do not actually have a functional Linux port and have officially discontinued support. The developers have noted the game does not even start for "most" people on Linux currently, but I expect it would at least launch in Proton.

**Comment by LuciusMartius** [comment]
Two more examples:
- Xenonauts: 'Community Edition' only available for the Windows version since Linux source code is with an external contractor.
- Gratuitous Space Battles: Native version lacks DLC support. There are very hacky methods of getting the Windows DLC to work with the native version, it would be a lot easier to just run the Windows version.

**Comment by Sanaki** [comment]
So unfortunately Nidhogg has proven there's an easy way for devs to kill linux support if this isn't fixed. Nidhogg has a "linux version", but it just has the OST in it, with no actual game files. Steam won't provide the option to use proton because a linux version exists, and attempting to bring the windows version into linux steam manually (with the app manifest as well) causes steam to delete the windows files on "launch" before once again complaining that it can't find an executable.

**Comment by BobbyWibowo** [comment]
HuniePop works better with Wine than its Linux version. It is as if the linux version was badly optimized. It could only be played smoothly after disabling desktop compositions

**Comment by dkhalansky** [comment]
I think there is a more general solution that would solve this issue as well as some others. An that is, allow a per-game setting of whether to use Steam Play--regardless of whether a game has a Linux port.

This way, one would be able not only to run Windows versions of Linux games but also avoid switching the "enable Steam Play for all titles" switch. An unfortunate consequence of using the switch is that one loses the ability to use filtering the list of games to only include SteamOS + Linux games to see which Windows games have been whitelisted.

**Comment by dkhalansky** [comment]
Also, BIT.TRIP RUNNER, Randal's Monday, and Blackguards 2 have files for Linux but without any executables and so can't be run.

**Comment by John-Gee** [comment]
I think we should be able to install both native and with Proton builds, that way it's easy to test which one behaves the best without having to redownload the game in case the first one tried was better, thus saving both ourselves and Valve bandwidth. Also, some games may behave better with their Linux build but lack multiplatform multiplayer and so having both builds may be useful.

A simple "Install Game" / "Install Game with Proton" in the right click menu and on the game page would do, though that might complicate things for new people so it might be better to hide that behind a setting.

**Comment by mirh** [comment]
> And the solution is, allow a per-game setting of whether to use Steam Play--regardless of whether the game has a Linux port

You first have to download the relevant game files to use steam play or not. 
Any option would have to be upstream, as soon as you press install.

**Comment by dkhalansky** [comment]
> You first have to download the relevant game files to use steam play or not.

Sure, but users can already control via settings the set of files that they download. It is already possible to choose the in-game language--which affects the set of files--as well as participation in beta builds. So I believe that a checkbox "Use Steam Play" in the Preferences window could work.

**Comment by legluondunet** [comment]
Another example: Enclave Linux port does not launch anymore for several months now.

**Comment by johndoe7000** [comment]
Linux native "Divinity: Original Sin" also does not work anymore with mesa + divos-hack shim:(

**Comment by miguev** [comment]
System Shock 2 doesn't have a real Linux port, but is instead a WINE+game that doesn't even start. Right now 2nd-5th top threads in https://steamcommunity.com/app/238210/discussions/ are about this problem.

**Comment by fredoche** [comment]
Clusterpuck 99 linux port has a framerate of 1 fps. I have hope that the windows version with proton vill perform better.

**Comment by rea987** [comment]
That can be a useful feature to play enable true cross-platform multiplayer for some game such as **Dawn of War II** and **Worms Reloaded**.

**Comment by mirh** [comment]
A similar drop down menu for the "Install" button would be nice, indeed (and you wouldn't need to add anything fancy in the normal right click options for games, which are already plenty). 

Now please.. I hope there's nothing left to discuss - so don't spam the thread with +1 examples.

**Comment by felixactually** [comment]
I echo Cpt-Derp: Garry's Mod has an awful Linux version. For some reason, it also doesn't work well with Wine either. Perhaps it would run well with Steam Play? I'm not sure, but it would definitely be a nice option to have, considering all the games mentioned so far. While most games with Linux support do work properly, it's nice to have a contingency plan in the form of "run this game with Steam Play instead of using the native version".

**Comment by cor3ntin** [comment]
Dying Light is also a rather broken Linux port (doesn't start with 18.04 + amd gpu). I, of course, would strongly prefer the game to be fixed, but the developer seems to have no intention to do so, and so Proton might be a workable alternative.

**Comment by rea987** [comment]
@Sanaki 

So, as you have mentioned in https://github.com/ValveSoftware/steam-for-linux/issues/5638#issuecomment-415886476, is that possible to co-exist both native and Proton version of the same game in the same Steam client via in-home streaming mechanism? Can you please elaborate adding Proton installation to in-home streaming preferences? Thank you!

**Comment by Sanaki** [comment]
Not exactly. In-home streaming allows you to stream games from a different computer entirely. I was saying that if the developers grant us this option, the menu to place the option in already exists in the client, and is explicitly for that sort of purpose. At the backend, they'll still need to figure out how they want to change the folder structure to support that decision, since currently proton and native versions install to the exact same location and directory name. Presumably it would be as simple as modifying the manifest for the game to indicate a secondary location, but I can't guarantee that would be the case. For all I know that would break something else hardcoded.

Either way, tl;dr: Right now no built-in method exists to run proton versions of games that have native versions available.

**Comment by mirh** [comment]
> I would like to extend the feature request with a few other cases

Those are yet other examples of why this is sensible. 
But it's already a feat Valve can unilaterally wrap other publishers products, I don't see what they could do without directly tampering with game files. 
(and it's probably OT)

**Comment by Sanaki** [comment]
He's just saying what the rest of this issue is saying: Allow us to install proton despite "native" versions. He's not suggesting that steam replace the wine version in the "native" port with proton.

**Comment by jplatte** [comment]
Another example: Cossacks 3. The linux port is horrible, it doesn't work **at all** with mesa.

**Comment by axredneck** [comment]
Another use case is Windows-only mods for Half-Life series which don't work with native HL versions. I'm ready to replace native HLs with their Windows versions because they work under Wine not worse.
p.s. Another game with broken Linux port is High Strangeness, and it works under Wine perfectly too.

**Comment by Mckol** [comment]
Dying Light doesn't work natively at all for me, would be great to be able to run it with proton.

**Comment by nstgc** [comment]
I'd like to point out the Topware ports in general which use outdated version of Wine. Fortunately the scripts that are used to run them can be changed to use system wine instead but even then Proton does a better job (or easier) of passing controller input to games. I've read that The Witcher 2 and Shadow of Mordor both suffer terrible performance penalties in their Linux ports. Overlord has controller input issues Age of Wonders III has a memory leak and the Linux version lacks the creation tools. Divinity: Original Sin EE also lacks creation tools in the Linux version.

**Comment by gnanini** [comment]
Radeon users sometimes have better performance on wine versions, specially with gallium nine and dxvk!

Saints Row IV is horrible on linux, plays nicely with dxvk

Chivalry Medieval Total war runs smoothly on wine with gallium nine. Linux version is missing some graphics details! (at least on mesa drivers)

Mad Max runs smoothly with dxvk

Shadow of Mordor could be better with dxvk

**Comment by jvalrog** [comment]
I think this feature would be useful to compare a linux port optimization. If the emulated version runs better, then the linux port is crap.

**Comment by mirh** [comment]
I think everybody has understood, agreed, concurred this is an important needed feature. 
Now please, stop with bumps.

**Comment by MartynHare** [comment]
This is very necessary as a feature, as some developers have games that don't actually work at all on Linux with their so-called Linux-native ports.  

BIT.TRIP Runner is a good example of this where there aren't even any game files downloading on the app I have on my Steam account.  I am currently waiting on Valve support via Steam Support ticket to get a resolution for this game, yet if Steam Play allowed me to pick the Windows version despite Steam claiming there's a Linux-native version, I'd be sorted.

**Comment by buenaventura01** [comment]
Hey, I have yet another case where this is useful: Sub Rosa ( https://store.steampowered.com/app/272230/Sub_Rosa/ ) has a native linux build that works well, BUT, I would like to play the windows only beta build. Letting me play the game via proton, even tho it has a native version, would solve this!

**Comment by aenertia** [comment]
Another example of a currently unusable Linux port - Civilization Beyond Earth. Native client segfaults on recent kernels due to changes in the Futex system ( see https://steamcommunity.com/app/65980/discussions/0/626329820749233064/?ctp=62 )

**Comment by Cpt-Derp** [comment]
@aenertia So much for "don't break userspace" if the kernel is the cause.

**Comment by JacobSvenningsen** [comment]
I just started the game fine from Linux 4.20 compiled off the DRM-next tree, and have previously not have troubles with that particular game, except right after it was released due to missing OpenGL support in Radeonsi.
It's definitely not a kernel problem

**Comment by aenertia** [comment]
Sure looks a lot like the old futex threads bug tho you have to admit.

What userspace are you using? I'm using 18.04 but with obiaf X updates for
mesa and 396.54 Nvidia from the proprietary updates ppa.

I tried several different 32bit runtime envs (including a 16.04 chroot) all
with the same result. Am open to suggestions. Also 4.20? I think you mean
drm-next head with 'potentially' included 4.20 patchsets. Mainline is had
only just gone to rc1 tag on 4.19. I run net-next on some things, and
unless there are some '100%' commits you know are going into 4.20 tagged
tree that's a little weird to call a next-tree for a release that isn't
fully tagged yet.

**Comment by Cytomax55** [comment]
This would be nice for games that dont have cross platform support like company of heroes 2

**Comment by Holston5** [comment]
I have created a workaround for this for the time being - https://github.com/Holston5/Native2Proton

**Comment by nsrosenqvist** [comment]
Another example game with bad Linux support that I'd prefer to be able to play through Proton instead: Move or Die

**Comment by sofakng** [comment]
More examples:  JackBox Party Pack games

**Comment by cassidyjames** [comment]
Cortex Command is yet another game that says it supports Linux, but actually has an empty download. As it is, I'm unable to play the game even though it has a working Windows version that has a [Gold rating](https://appdb.winehq.org/appview.php?iAppId=5211) with Wine.

**Comment by wirosl** [comment]
I want to second this feature.

On my end, its Shroud of the Avatar: Forsaken Virtue. 
Due to unresolved bugs in the Linux client, it never recieved the version upgrade in Unity that its Windows counterpart did, thus missing out on some of the performance fixes. In addition, someone made the thoughtful decision that after deciding to make and support the game for multiple operating systems, to go ahead and use Windows-only plugins for some in-game features like being able to use maps, and then just disabling those important features on the other platforms...

So yea... just because a game already has a Linux port doesn't mean it wouldn't heavily benefit from having the ability to use the Windows version on the Proton back-end.

**Comment by genghisnico13** [comment]
I would like to have this feature too, Rocket League Linux port doesn't have controller vibration(rumble) implemented and if it wasn't implemented in the last 3 years I doubt it will ever be.
Also maybe the workshop maps could work with proton, currently they don't.

**Comment by pchome** [comment]
> ... yet another game ...

[Virtual Programming eON](https://pcgamingwiki.com/wiki/Company:Virtual_Programming#Games_ported_to_macOS_.28formerly_OS_X.29) ports.
Maybe [some other third-party ports](https://pcgamingwiki.com/wiki/Company:Aspyr_Media#Games_ported_to_macOS_.28formerly_OS_X.29) .

Or even **all** third-party ports could potentially benefit from Proton. I think no more examples needed.

**Comment by monsterbitar** [comment]
Eador: Masters of the broken world "has linux support", but in several years I've yet to managed to even start it.

Kingdom of loot started out with linux support, but due to real-life issues couldn't keep the support, leaving linux users with a windows-only game.

I reckon about 10% or so of all my linux-titles are functionaly broken, and I would like to be able to use proton as an option for these games. As a user already suggested, having the option at install time to install native or install steam-play would make sense.

**Comment by nstgc** [comment]
@Tele42 The problem is that this puts it on ValvE, who are taking an increasingly hands off approach. The fact that there are games that haven't worked in years on the store is evidence of why it can't be left up to the devs nor ValvE.

I do see the logic in your argument, especially when you consider the possibility of an influx of new Linux users who will be dead set on trying everything out and breaking everything in spectacular ways.

**Comment by John-Gee** [comment]
@Tele42 I don't think this is good, simple because of a weak assumption: that a port is either better or worse than with Proton, while it could be both on different things. 

Just look at the Feral ports, they are usually of good quality, but often don't allow multiplayer with non nix platforms, where Proton would (assuming the games run of course).

Now we can also think that what you're suggesting is pretty similar to what Valve is doing with the whitelisting, so yeah sure, let Valve whitelist games that already have ports, yet broken ones, while still giving us the choice at the end.

**Comment by daasuharo** [comment]
How would this affect payout? Someone using proton on Windows games are counted as linux sales so wouldn't some third parties that worked on a native linux port possibly get paid even though you could have played the game only/mostly through proton during the first two weeks?

**Comment by John-Gee** [comment]
For what @daasuharo said, I think a simple way out is to track the publisher of the build ran the most during the first 2 weeks, and act accordingly. Of course that's probably very naive since I have no contractual/financial information.

Also I doubt the issue is new, people could already run Steam games in Wine before, even games that had native ports... So I'm guessing Valve already knows what to do there.

**Comment by daasuharo** [comment]
I don't think wine plays counted as linux sales unless it was an official linux port using wine. You still had to use the windows steam client. Proton counting as linux sales is a new thing I believe.

**Comment by John-Gee** [comment]
Right but that's a technicality, my point was that they most likely already thought about the issue previously. The fact that Proton counts as Linux is why I suggested looking at the publisher instead of the platform :)

**Comment by jarrard** [comment]
I'd like this feature also, hopefully something like what Sanaki suggested can be done. 
Warthunder and Tomb Raider 2013 could benefit from this function for example, but there are plenty more (arma3). Basically the native OpenGL versions are subpar atm compared to DXVK/Proton option.

**Comment by jorammer** [comment]
I would like the feature in question because the Linux version of Everspace does not have VR support. Only the Windows version does.

**Comment by Ripolak** [comment]
I would absolutely love to see this feature. Today I tried running Civilization V but it has known problems with many mods that cause it to crash on the Linux version. I strongly believe that running the Windows version will solve many, if not all the problems of this kind.

**Comment by Kryuko** [comment]
A new example is Ark survival evolved.
The linux version is still on opengl 3, with a lot of bugs, missing effects, and glitches.

**Comment by Kryuko** [comment]
I ask for a feature that let us report games with a broken linux version, so they can be removed from the store. Is not a good thing for a new linux user to see a game, buying it, and then understand that he cant play the game because the dev gave them a broken game.

Maybe then send an email to the game developer (after x reports) to let him know that his linux version of the game has been removed from the store.

Thank you.

**Comment by jarrard** [comment]
The developer of ARK could move to Vulkan API, but they spent all their money on assets addons etc so who knows if they even have the money to pay someone to do that. Unreal4 engine supports vulkan which is what it uses..

**Comment by Kryuko** [comment]
> The developer of ARK could move to Vulkan API, but they spent all their money on assets addons etc so who knows if they even have the money to pay someone to do that. Unreal4 engine supports vulkan which is what it uses..

They announced in 2016 that they wanted to use vulkan, we are at the end of 2018.
We have waited enough, let us play the windows version!

**Comment by sirkon** [comment]
Civilization 5 does have linux support, but:

1. Its performance is a bit underwhelming: my old 8800gtx can't handle smooth framerates even at lowest settings, although everything is smooth with integrated videochip of Core i5 4670k in Windows.
2. Many mods just don't work in Linux version

**Comment by dragonfly-lover** [comment]
I would gladly try GRID autosport and Dirt Rally under proton because Feral Ports at that time were using OpenGl, and finally FPS are below expectancies

**Comment by truboxl** [comment]
To add on a bit on DiRT Rally,
there are people out there with 300+ hours worth of game saves on the Windows version, but later they find out that their saves are not cross platform compatible with the Linux version and have to nuke it or disable Steam Cloud to play on Linux.

I think Feral did what they could for the port but this really sucks if gamers have to start over...

Maybe other games have this kind of problem too...

**Comment by Cytomax55** [comment]
I have no idea how much effort it would take to allow people to play games ported to linux with steam play but my assumption is not much.... what I am scared of is valve doesn't want to use this software to replace linux ports and is scared of the message that would send to developers that actually take the time to port to linux... is this feature not added for a technical reason or a political reason?

**Comment by veikk0** [comment]
Another reason to allow this is potential power savings and reduction of GPU fan noise. For example, Dead Island has a Linux port that is functional but has extremely limited graphical options in comparison to Windows. No way to do anti-aliasing or set a framerate limit. As a result the game renders only at the monitor's native resolution, which any decent GPU will be able to crank out at 100+ fps in the case of a 1080p image. That's just excessive unless you have a 120/144 Hz screen and wastefully runs the GPU at 100% for no real benefit.

**Comment by Plaque-fcc** [comment]
+1 to this feature.

**Comment by linuxgaming** [comment]
I hope this issue is addressed before we see more of [this](https://steamcommunity.com/games/474210/announcements/detail/1684800028906686550)

**Comment by s-m-k** [comment]
@linuxgaming Regarding BUTCHER, we changed our mind about that. We'll revert to the old configuration, That said, we hope that this feature is implemented by Steam fairly soon.

**Comment by Holston5** [comment]
> 
> 
> @linuxgaming Regarding BUTCHER, we changed our mind about that. We'll revert to the old configuration, That said, we hope that this feature is implemented by Steam fairly soon.

That's good to know.

**Comment by wsanders99** [comment]
Another good example is the original Dungeon Defenders. The Linux and macOS clients are stuck on version 7.48 while the Windows version still receives updates. Saves also seem to be broken between versions, sometimes corrupting them.

**Comment by codedcosmos** [comment]
Probably will also be a solution to the buggy controller support on Linux for risk of rain.

**Comment by MartynHare** [comment]
I support this feature, sometimes games have mods that are exclusively made for the Windows variant by people that can't or don't have the time to grok Linux.  Also, Valve rocks for actually listening to my Wine suggestion from 2017 that people said would never happen :-P

On a side note:  @s-m-k Just remember if you get stuck, the Linux community is more than willing to help with your Linux port, just as we want to help Valve make the back catalogue of older Windows-only games "just work" despite not profiting personally from the endeavour.

**Comment by SasakiSadako** [comment]
i really hope there will be this Option too someday.

Im happy to play native Ports like Tomb Raider, but other games, are more buggy, or you cant play multiplayer with Windows Users.

Wish to play the original Version of Shogun 2 on my Linux System in Steam to use the Workshop.

**Comment by steffenWi** [comment]
Another example would be Cossacks 3. The Linux version either crashes upon selecting a profile, freezes or - if one manages to get it to run - doesn't render water at all.

**Comment by jtiemer** [comment]
The Banner Saga also has a borked Linux Version. Steam Play for borked native titles would be most appreciated.

**Comment by VictorVolovik** [comment]
Total War series don't have cross-platform multiplayer, despite having some Linux versions now (Total War Warhammer 1) and coming (Total War Warhammer 2 - devs already have confirmed that there is no cross-platform multiplayer support planned).

**Comment by dpanter** [comment]
> Dying Light also has a borked linux version that they have refused to try and fix since 2016. (white loading bar then crash). The only way to get it to work is to either run steam in a snap on an older version of the Linux Steam integration snap libraries or to run it specifically on ubuntu 16.04 (doesnt run on ubuntu 18). I would love to be able to play this game that I've owned for 3 years and been unable to play :/

I cannot agree with this statement. The Linux version is definitely not borked.
It may not run for you, but it runs well for many Linux gamers. I'll admit it is far from perfect and could be better optimized. Trying to get it to run on Mesa drivers has been a long, painful journey. The white loading bar -> crash is not exclusive to Linux either. I've seen way more problem reports regarding Windows 10 than with Linux...
But borked? Naw, man. Flawed? Yeh, man.

Btw, I hear it runs well with Native2Proton. Haven't tried myself since it runs great natively for me.

**Comment by GloriousEggroll** [comment]
@dpanter what OS are you on. I've been on Arch + mesa and it's never run correctly for me even with overrides. The only time I've ever been able to get it to run was in snap using solus snap runtimes and solus LSI snap libraries + the overrides.

**Comment by GloriousEggroll** [comment]
as far as I recall it's always worked on nvidia regardless of distro. it was fixed in mesa a while back, but its broken again due to a glibc segfault

**Comment by smdias65** [comment]
Ticket to Ride has this issue. The devs abandoned the Linux port with beta version 2.3.1; the Windows version is 2.5.5, but we can't install it through Linux/Proton. The Linux beta frequently crashes.

**Comment by TomTheGeek3** [comment]
Arma III even though it has a "native" port, is not up to par with the Windows version (as in up to date.)

I'd like to see a option to install the windows version regardless if it has a native version or not.

**Comment by 2E0PGS** [comment]
Yep I need this so I can try running Gmod via proton and see if it works better with audio issues/glitches. Related to issue: https://github.com/Facepunch/garrysmod-issues/issues/3090

**Comment by hartmark** [comment]
+1 My Vega 64 is unable to run Civ6

**Comment by kogorek** [comment]
+1 Cortex commander has broken linux port(literally empty directory), so I need to launch virtual machine to play it

**Comment by JOndra91** [comment]
@kogorek Cortex Command has a beta branch which you have to unlock with code `rocket100000`.

Then you have to switch to `development` branch (which says OSX but it's really a Linux version). But as it is, it shouldn't be advertised as having a Linux version. And it's probably still more stable on Windows.

**Comment by henk717** [comment]
+1 there are a few games with inferior Linux versions which may work better under proton.
This counts for ARK but potentially also HL2DM since HL2DM has some linux specific crashes.

**Comment by Kryuko** [comment]
I think we need, at least, a system that allow us to report not working games on Linux, so valve can remove them from the store and there will no better false advertise on some games.

**Comment by 8BitCerberus** [comment]
Adding my +1 to this. Just came across an issue last night that I wouldn't have thought would ever be an issue, until now.

I got American Truck Simulator on last month's Humble Monthly Bundle so I figured I'd try it out. The game has a native Linux port, and an official Steam Controller binding, so that was pretty cool to see. Except the Steam Controller binding doesn't work. It's a mix of keyboard & mouse + gamepad. So I tried going full gamepad. No good. Full keyboard & mouse with the Steam Controller is fine though (but then you don't get analog steering and acceleration). 

I even tried using sc-controller which has worked for everything that I can't get to work through Steam (which is very little, and typically only non-Steam games), still no good. So I started digging, and it seems the game does in fact work fine with the Steam Controller... if you're playing on Windows. After some time I came across this Reddit thread: https://www.reddit.com/r/SteamController/comments/6n5bez/support_steam_controller_with_euro_truck/

It seems the Windows version of the games works just fine with any Xinput capable controller, and even works great with mixed Xinput + keyboard & mouse, based on the official Steam Controller bindings. However, the Linux version apparently only works specifically with Xbox 360 controllers, not any gamepad capable of Xinput. I haven't yet, but I'm going to try the workaround in that thread to spoof X360 vendor ID, see if that gets me going with Steam Controller, but I suspect this wouldn't even be an issue if I could just run the Windows version of the game w/ Steam Play.

**Comment by Kryuko** [comment]
> Adding my +1 to this. Just came across an issue last night that I wouldn't have thought would ever be an issue, until now.
> 
> I got American Truck Simulator on last month's Humble Monthly Bundle so I figured I'd try it out. The game has a native Linux port, and an official Steam Controller binding, so that was pretty cool to see. Except the Steam Controller binding doesn't work. It's a mix of keyboard & mouse + gamepad. So I tried going full gamepad. No good. Full keyboard & mouse with the Steam Controller is fine though (but then you don't get analog steering and acceleration).
> 
> I even tried using sc-controller which has worked for everything that I can't get to work through Steam (which is very little, and typically only non-Steam games), still no good. So I started digging, and it seems the game does in fact work fine with the Steam Controller... if you're playing on Windows. After some time I came across this Reddit thread: https://www.reddit.com/r/SteamController/comments/6n5bez/support_steam_controller_with_euro_truck/
> 
> It seems the Windows version of the games works just fine with any Xinput capable controller, and even works great with mixed Xinput + keyboard & mouse, based on the official Steam Controller bindings. However, the Linux version apparently only works specifically with Xbox 360 controllers, not any gamepad capable of Xinput. I haven't yet, but I'm going to try the workaround in that thread to spoof X360 vendor ID, see if that gets me going with Steam Controller, but I suspect this wouldn't even be an issue if I could just run the Windows version of the game w/ Steam Play.

I have the same problem with Dying Light, the game is so good and it runs fine, but i can't use the steam controller with this game, THAT IT WORKS ON WINDOWS.

And this is a real problem for me, because i have tendonitis at both my wrists so i can't play anything with mouse and keyboard.
In the past the steam controller worked fine with this game.

**Comment by leillo1975** [comment]
About ats and ets2, i can use steam controller without problems with sc-controller:
https://github.com/kozec/sc-controller

With Dying Light I have no problems and it runs well

**Comment by Schlesiger** [comment]
Civ5 mods often require Windows to run. I'd like to install the Windows version of Civ5 to install mods such as [the Community Patch](https://forums.civfanatics.com/threads/community-patch-how-to-install.528034/).

**Comment by hartmark** [comment]
I found this little tool to download windows games and run them with proton. I haven't tried it out so much yet though, but it looks promising.

**Comment by dearlygparted** [comment]
The Banner Saga (appid 237990) used to have a Linux port. The developers have dropped Linux support, and the SteamOS logo was removed from the Steam Store page. People who already own the game could still play it on Linux. 

However, the game no longer works and developers are refusing to fix it. This is exactly why we need to be able to use Steam Play for "native" Linux games.

**Comment by sinedoOo** [comment]
Linux native "Divinity: Original Sin" does not work with current mesa. We really need this option!

**Comment by Sirmentio** [comment]
I feel as though that Native2Proton program works pretty okay for what it is, but it doesn't seem to allow you to update existing games. I feel as though we need to have this officialy rather than through a separate program.

**Comment by Cytomax55** [comment]
I wonder why Valve hasnt implemented this feature yet 
I wonder if this is due to technical reasons or business reasons

If they allow you to play native linux games through proton that might send the wrong message to devs and they might stop porting to linux and just target vulkan for windows and shoot for proton compatibility.

Very sticky situation for Valve but regardless i would like to have this option!

**Comment by xDShot** [comment]
Business and licence agreements, I guess. Some games have their own publishers on Linux platform specifically.

**Comment by supertin** [comment]
+1 for the choice via a dropdown....
BUT - consideration needs to be made for handling files that pre-exist for the alternate mode. Steam could either:
a) treat the different platform modes as different library storage entries.
b) delete the Linux version and download the Windows one (or vice versa as required).
c) do some proper de-duplication, and only download and store the files that are actually different.

A would waste a lot of disk space, downloads and time.
B would at least reduce the space requirements, but depending on the game, may still waste a lot of time and downloads.
C would give the best user experience, with limited wasted space and download, and no requirement to re-download any files when flipping between platforms. But it would also be hardest to implement.

**Comment by JoneKone** [comment]
> +1 for the choice via a dropdown....
> BUT - consideration needs to be made for handling files that pre-exist for the alternate mode. Steam could either:
> a) treat the different platform modes as different library storage entries.
> b) delete the Linux version and download the Windows one (or vice versa as required).
> c) do some proper de-duplication, and only download and store the files that are actually different.
> 
> A would waste a lot of disk space, downloads and time.
> B would at least reduce the space requirements, but depending on the game, may still waste a lot of time and downloads.
> C would give the best user experience, with limited wasted space and download, and no requirement to re-download any files when flipping between platforms. But it would also be hardest to implement.

B is best.

**Comment by Rabcor** [comment]
Man this is a highly requested feature.

**Comment by xaxazak** [comment]
Just want to add that some games have multiple launch options (launch game vs launch editor, etc). You might want to run the game natively but the editor in proton. (Eg: Overload _might_ be such a game (I can't test editor on proton)).

So +1 for allowing side-by-side (or deduplicated, as mentioned above) proton and native installs.

**Comment by sunderkeenin** [comment]
Dungeon Defenders is kind of a problem. There are some major bugs, and one dealbreaking downside.
Major bug - mana physics don't work
Dealbreaking downside - you're on a different version from Windows users, meaning you cannot play with your friends. The game runs great via winesteam on lutris though, and would make a great addition to my personal steamplay list if I could install the windows version and run it via proton.

**Comment by frostworx** [comment]
would be also mandatory for several VR games:
at least
Tinertia
Rise of the Tomb Raider
Dirty Rally 
don't have VR support on linux (or I haven't found a way yet for TR and DR)

**Comment by sorin255** [comment]
Saints Row the Third runs decently on Linux but is missing LAN/Online play. It also does this weird thing where it loads textures as low res and then changes them to high when the camera focuses on them (which takes a second or two to load). Civ 5 does the same thing. Not unplayable but not pretty.

**Comment by cirospaciari** [comment]
ARK: Survival Evolved never update the game in Linux/Mac and we have a broken experience. Textures missing and bugs are common. If we just have the option to play with steamplay that will be solved.

**Comment by NWalker1208** [comment]
Event[0] has a very rough Linux beta, so running with Proton would be much better.

**Comment by mhess126** [comment]
PLEASE add this.  With all respect to the linux porters, the native version of Rocket League is inadequate.  I cannot play it with stable frame pacing at any resolution / settings, and on Windows I can play smoothly at 1440p with all settings checked.  There's something fundamentally broken in that port, which is too common with native linux games.

**Comment by mcgravier** [comment]
WarThunder would significantly benefit from this - They regressed from OpenGl 4.x to 3.x recently causing significant performance drop on their native client - DXVK currently offers far better experience

**Comment by codedcosmos** [comment]
Rocket league worked quite well for me, nearly maximum settings on RX 570 1080p. @NWalker1208

**Comment by frostworx** [comment]
Distance has native linux build, but without VR support...

**Comment by JoneKone** [comment]
> Rocket league worked quite well for me, nearly maximum settings on RX 570 1080p. @NWalker1208

Rocket league has Native support and is OpenGL game

**Comment by JoneKone** [comment]
On most Battle Eye utilizing games it's battle eye that is the issue.. They think that cheaters will start using Linux and they can't control them.

**Comment by Oliver-1xok** [comment]
While Proton is on Windows level for many games, some ports are not really maintained anymore. I don't want to criticize anyone for it. Linux is only a small market. But why not take Proton if it works better? We Linux users can cope with such freedom of choice. :)

**Comment by radixs** [comment]
yeey, please add that feature. My favourite game Creeper World 3 runs lousily on Ubuntu 18.04 but super-fast on windows. I have a feeling running it on proton would help greatly.

**Comment by Skodra** [comment]
WE cant play CIvilization VI Linux version multiplayer with the rest of the players. Mac and WIndows users can play between themselves though. It would be nice if I could launch windows version because I only play multiplayer with this game.

**Comment by Lyle-Tafoya** [comment]
There is no need to mention every single game in which you would like to try the Windows version on Linux. All this does is unnecessarily notify everyone who is watching this issue. I want to be notified when there is a relevant update to this feature request. Not every time someone finds another game they want to add to the list.

**Comment by kode54** [comment]
FEZ: The Linux port is unplayable/broken: The screen doesn't scroll around as you move, thus you can end up walking right off the edges of the viewport and end up blind to your own actions. The viewport seems to lock whenever you enter a new area.

**Comment by Saroumane** [comment]
Another example : "The Next Penelope", Linux version is officially discontinued and broken, and we can't install steamplay proton version.
https://www.protondb.com/app/332250

**Comment by HikloLime** [comment]
Another example would be The witcher 2 linux native, which is the most crash prone and broken linux port I've ever played. Would be lovely to play such a great game without crashes!

**Comment by kode54** [comment]
I need to add to my previous comment regarding FEZ for Linux, it does work, at least on Arch and Plasma, possibly due to using an Xbox360 controller, or at least having the controller properly activated prior to launching.

**Comment by nzar2k** [comment]
Jesus Christ, enough mentioning your buggy games!
We **don't** want to know which game exactly runs bad using native port in this thread!

**Comment by radixs** [comment]
> Jesus Christ, enough mentioning your buggy games!
> We **don't** want to know which game exactly runs bad using native port in this thread!

Mentioning buggy games seem to be relevant to all those people that commented here. Who do you have in mind by using the word "we"? Yourself and?

**Comment by mati865** [comment]
This issue tracks option to force Proton over native port.
Listing dozens of broken games is off topic and unnecessary spam.

---
It'd be much cleaner if you opened organised issue tracking broken native games with content like:
- [ ] Game 1 - doesn't start
- [ ] Game 2 - crashes every 15 minutes

Maybe Valve could enable Proton for them if they cannot do it for everything (due to licensing or whatever else).

**Comment by snqlby** [comment]
We have this wonderful site to report game experience with Proton:
https://protondb.com
First of all, this place is not intended for reports about non-working games. Implementing a feature from the topic requires serious discussions, as it may lead to a weakening of the development of games for Linux in the future, following the path of least resistance

**Comment by frostworx** [comment]
even more annoying than the reports is this discussion, which is not less off-topic but more active
*muting this thread* pretty sure I won't miss the fix for this issue

**Comment by nstgc** [comment]
> We have this wonderful site to report game experience with Proton:
> https://protondb.com
> First of all, this place is not intended for reports about non-working games. Implementing a feature from the topic requires serious discussions, as it may lead to a weakening of the development of games for Linux in the future, following the path of least resistance

We aren't talking about Windows games that don't work in Proton. Rather we are talking about supposedly Linux games which don't actually work, or not work properly. ProtonDB has nothing to offer in this case.

**Comment by supertin** [comment]
This issue is about the Steam *client* itself. As has been mentioned several times... STOP LISTING GAMES THAT DON'T WORK.

If you have problems with the native Linux version of a specific game, take it up with the game vendor. Listing it here will NOT help anyone.

**Comment by SwooshyCueb** [comment]
There's a crazy machines game that has an empty linux depot, so the steam client thinks there's a linux version. Can't use proton with it as a result.

**Comment by grinapo** [comment]
Do I see right that in this last 4 months and hundreds of comments there was absolutely no reaction from Valve™ here? 
Also, there seem to be no hack or tweak to make this forced upon the client, even if by fiddling with config files? 
I've tried to browse through the comments but apart from the original issue report I haven't noticed any progress or solutions, am I observing right?
(Purposefully I skip mentioning the games I'd like to run. ;-))

**Comment by Saroumane** [comment]
> Do I see right that in this last 4 months and hundreds of comments there was absolutely no reaction from Valve™ here?
> Also, there seem to be no hack or tweak to make this forced upon the client, even if by fiddling with config files?
> I've tried to browse through the comments but apart from the original issue report I haven't noticed any progress or solutions, am I observing right?
> (Purposefully I skip mentioning the games I'd like to run. ;-))

There is a hack called "Native2Proton" which uses Steamcmd : 
https://github.com/Holston5/Native2Proton
(I did not try it)

**Comment by adelpozoman** [comment]
a lot of feral interactive ports have bad performance and launching at ubuntu 18.04 with amdgpu gives a warning about unsupported system that can affect performance. wtf.

**Comment by GloriousEggroll** [comment]
> Does this issue need everyone to list games they can't play? You can just subscribe.
> 
> Dying light on debian/Nvidia has always worked since release for me and this isn't really an issue for discussing which games need this option.
> 
> We just need a drop down to choose which platform to install like with streaming games from another pc.

Dying light's problem with the native port is libglvnd. If you compile mesa without libglvnd it works fine. There is an unresolved open issue on it:
https://github.com/NVIDIA/libglvnd/issues/165

**Comment by Cytomax55** [comment]
All the feral games I have played have played well coh 2 And tomb raider... I get the same warning but games play fine... Still it would be nice to be able to choose if I play native games for Linux or the Windows using proton

**Comment by GloriousEggroll** [comment]
> Linux native "Divinity: Original Sin" also does not work anymore with mesa + divos-hack shim:(

shim works fine here, just ran it yesterday, the shim is for the definitive edition

**Comment by kapcom01** [comment]
I have played Life is Strange Before The Storm for 7 hours with SteamPlay and then I got an update to the native version. The native version does not support Intel graphics.. I just can't play the game anymore.. I would love to see an option to go back to the SteamPlay version.

**Comment by JoneKone** [comment]
> Jesus Christ, enough mentioning your buggy games!
> We **don't** want to know which game exactly runs bad using native port in this thread!

Absolutely no comment has been said from Valve on this issue. How do you expect this thread to stay alive if not for ppl who mention their dissatisfaction to the current state of this issue? It will just drop to the bottom of the list and will be forgotten and ignored. We all know why this issue is, it's cause developers/publishers don't want to allow us to use this kind of tool/option.

Only by keeping this alive we can start seeing results.

I mean there is at least 1 dev who dropped the linux port in favor of Proton "emu" being far better option for gamers. I personally believe it was this thread that made it happen.

**Comment by toojays** [comment]
> We all know why this issue is, it's cause developers/publishers don't want to allow us to use this kind of tool/option.

If don't follow this logic. Surely the publishers and first-party developers want the users to play with whatever setup gives them the best experience? Maybe there's a bit of an accounting glitch that can occur when a third-party has ported a game with the expectation of a percentage of sales reported as played on Linux by Valve, but I don't see that stopping Valve.

> Only by keeping this alive we can start seeing results.

I'm wondering if it shouldn't be the other way around. Instead of everyone piling on to this thread listing their pet games that run well in Proton but poorly in their native builds, file whitelist requests for such games instead. Sure that's not the same as being able to select Proton vs Native from the UI, but it might work better for those that care about individual games.

**Comment by nstgc** [comment]
> Hello @toojays, until we hear from a Steam or Proton dev one way or another, whitelist requests for native linux games will not be considered. Fragmentation of the discussion only adds pointless overhead to the issue trackers.
> 
> Regardless of the amount of activity here, the interested devs are aware of this request.

Is this really an issue for the developers? Seems like this is the sort of thing that the legal department would need to handle. I mean, if anyone in this thread would know its you, so I'm asking out of genuine curiosity.

**Comment by mirh** [comment]
> whitelist requests for native linux games will not be considered

This is not about making *any* "proton wrapping" the default, or overriding any dev preference. 
This is just about (somehow) **not** hiding it when native is also available.

**Comment by JoneKone** [comment]
> > We all know why this issue is, it's cause developers/publishers don't want to allow us to use this kind of tool/option.
> 
> If don't follow this logic. Surely the publishers and first-party developers want the users to play with whatever setup gives them the best experience? Maybe there's a bit of an accounting glitch that can occur when a third-party has ported a game with the expectation of a percentage of sales reported as played on Linux by Valve, but I don't see that stopping Valve.
> 
> > Only by keeping this alive we can start seeing results.
> 
> I'm wondering if it shouldn't be the other way around. Instead of everyone piling on to this thread listing their pet games that run well in Proton but poorly in their native builds, file whitelist requests for such games instead. Sure that's not the same as being able to select Proton vs Native from the UI, but it might work better for those that care about individual games.

That is a naive answer and question you are making there.

**Comment by grinapo** [comment]
There is already an option to enable Proton manually for the games. 

What are the specific reasons not to make it possible to **manually switch** between native and proton version, if both available? Is it a legal or a technical problem, or something else?

**Comment by radixs** [comment]
> There is already an option to enable Proton manually for the games.
> 
> What are the specific reasons not to make it possible to **manually switch** between native and proton version, if both available? Is it a legal or a technical problem, or something else?

I would bet that it's just that we linux master race are only a very small community. We have a very low priority. On top of that number of cases where protoned games work better than native versions is narrowing that drastically even more. 

So there you go - this request sits on a very low priority list. If you want to change that I guess convince more people in the world to use linux :D 

On the other hand if Valve is so supportive with gaming on linux this should have been solved the moment they made the proton available. Why didn't they add this convenient option right at the start? Might valve be having any interest in keeping native ports alive? Maybe they own shares of some other company that is making such ports?

Anyways it seems like a simple job to add the option to choose proton also for native linux games. Just add bunch of 'if' statements in various points without changing existing code. At least it smells that way.

**Comment by codedcosmos** [comment]
Oddly this is one of the biggest issues with proton, and is relatively easy to solve (at least from my perspective, there may be underlying issues that are hard to circumvent).

I just wish valve would communicate. At least say if they are thinking about it or if its coming sometime in the future.

**Comment by LiamDawe** [comment]
Added to the latest beta: https://steamcommunity.com/groups/SteamClientBeta#announcements/detail/1703951108827819236

**Comment by kisak-valve** [comment]
Closing as implemented per "Added the ability to force-enable Steam Play in per-title properties, including for native games" in the [2019-01-17](https://steamcommunity.com/groups/SteamClientBeta#announcements/detail/1703951108827819236) Steam client beta update.

Note: I'm going to lock this thread to avoid spam to the participants.
